Firebase/DynamicLinks/Utilities/FDLUtilities.m

```diff
@@ -212,10 +212,12 @@ BOOL FIRDLIsURLForWhiteListedCustomDomain(NSURL *_Nullable URL) {
             ([urlStr characterAtIndex:domainURIPrefixStr.length] == '/' ||
              [urlStr characterAtIndex:domainURIPrefixStr.length] == '?')) {
           // Check if there are any more '/' after the first '/' or '?' trailing the
-          // domainURIPrefix.
+          // domainURIPrefix. This does not apply to unique match links copied from the clipboard.
+          // The clipboard links will have '?link=' after the domainURIPrefix.
           NSString *urlWithoutDomainURIPrefix =
               [urlStr substringFromIndex:domainURIPrefixStr.length + 1];
-          if ([urlWithoutDomainURIPrefix rangeOfString:@"/"].location == NSNotFound) {
+          if ([urlWithoutDomainURIPrefix rangeOfString:@"/"].location == NSNotFound ||
+              [urlWithoutDomainURIPrefix rangeOfString:@"?link="].location != NSNotFound) {
```

Comment:
Does the `?link=` need to be the first thing that appears after the URI prefix? This seems like it would match something like:

`https://a.firebase.com/mypath/extra/stuff/here?link=https://www.google.com&test=1`
with a domainURI prefix of `https://a.firebase.com/mypath/` - or am I missing something?

Reply:
There are some inconsistencies for long links created from within the app preview page (e.g. extra /. Also, am not sure yet if there are any other). Given that we want to primarily check for the validity of the domainURIPrefix, was a bit more permissive on these. For developer created incorrect links, the backend would reject those.
What says?

Reply:
Okay, that sounds reasonable I think.